### DISCUSSION TITLE
getting error {error,{"no such file or directory", "rabbitmq_delayed_message_exchange.app"}} in rabbitmq docker image

### DISCUSSION THREAD

**Discussion by KamranShahid** [rabbitmq/rabbitmq-server#5130] [Other]
BOOT FAILED
===========
Error during startup: {error,{"no such file or directory", "rabbitmq_delayed_message_exchange.app"}}

Kernel pid terminated (application_controller) ({application_start_failure,rabbit,{{"no such file or directory","rabbitmq_delayed_message_exchange.app"},{rabbit,start,[normal,[]]}}})

Crash dump is being written to: /var/log/rabbitmq/erl_crash.dump...done

more details are

rabbitmq_delayed_message_exchange   

   Plugin doesn't support current server version. Actual broker version: "3.10.5", supported by the plugin: ["3.7.0-3.7.x", "3.8.0-3.8.x"]

What should be the remedy

**Comment by michaelklishin** [comment]
The error message tells you exactly what the issue is: you have downloaded a version that is not compatible with RabbitMQ 3.10.x.

You have to use the most recent releas of that plugin.

**Comment by mkuratczyk** [comment]
Also worth noting that the "official" RabbitMQ image (https://hub.docker.com/_/rabbitmq) doesn't include the delayed message exchange plugin at all.

**Comment by KamranShahid** [comment]
i have simply downloaded rabbitmq:latest docker image.
isn't it suppose to handle there automatically?

**Reply by mkuratczyk** [reply]
`rabbitmq_delayed_message_exchange` plugin is not shipped in RabbitMQ itself (has to be "installed" separately) and is not present on the `rabbitmq` image:
```
$ docker run -it rabbitmq:latest -- rabbitmq-plugins list
Listing plugins with pattern ".*" ...
 Configured: E = explicitly enabled; e = implicitly enabled
 | Status: [failed to contact rabbit@7f1a92425997 - status not shown]
 |/
[  ] rabbitmq_amqp1_0                  3.10.5
[  ] rabbitmq_auth_backend_cache       3.10.5
[  ] rabbitmq_auth_backend_http        3.10.5
[  ] rabbitmq_auth_backend_ldap        3.10.5
[  ] rabbitmq_auth_backend_oauth2      3.10.5
[  ] rabbitmq_auth_mechanism_ssl       3.10.5
[  ] rabbitmq_consistent_hash_exchange 3.10.5
[  ] rabbitmq_event_exchange           3.10.5
[  ] rabbitmq_federation               3.10.5
[  ] rabbitmq_federation_management    3.10.5
[  ] rabbitmq_jms_topic_exchange       3.10.5
[  ] rabbitmq_management               3.10.5
[e ] rabbitmq_management_agent         3.10.5
[  ] rabbitmq_mqtt                     3.10.5
[  ] rabbitmq_peer_discovery_aws       3.10.5
[  ] rabbitmq_peer_discovery_common    3.10.5
[  ] rabbitmq_peer_discovery_consul    3.10.5
[  ] rabbitmq_peer_discovery_etcd      3.10.5
[  ] rabbitmq_peer_discovery_k8s       3.10.5
[E ] rabbitmq_prometheus               3.10.5
[  ] rabbitmq_random_exchange          3.10.5
[  ] rabbitmq_recent_history_exchange  3.10.5
[  ] rabbitmq_sharding                 3.10.5
[  ] rabbitmq_shovel                   3.10.5
[  ] rabbitmq_shovel_management        3.10.5
[  ] rabbitmq_stomp                    3.10.5
[  ] rabbitmq_stream                   3.10.5
[  ] rabbitmq_stream_management        3.10.5
[  ] rabbitmq_top                      3.10.5
[  ] rabbitmq_tracing                  3.10.5
[  ] rabbitmq_trust_store              3.10.5
[e ] rabbitmq_web_dispatch             3.10.5
[  ] rabbitmq_web_mqtt                 3.10.5
[  ] rabbitmq_web_mqtt_examples        3.10.5
[  ] rabbitmq_web_stomp                3.10.5
[  ] rabbitmq_web_stomp_examples       3.10.5
```

**Comment by KamranShahid** [comment]
I simply pulled one older image rabbitmq:3.8.34 instead of the one's i am trying before (rabbitmq:3-management-alpine and rabbitmq:latest) and didn't see that error.

**Reply by mkuratczyk** [reply]
The error is about incompatibility between an old version of the plugin and new version of RabbitMQ. Using an old version of RabbitMQ is hardly a solution - 3.8 will be out of support in a few months.

**Reply by michaelklishin** [reply]
In a few weeks, in fact.

**Comment by KamranShahid** [comment]
I am simply pulling image and doing nothing after. I thought it should be handled in the image

**Reply by mkuratczyk** [reply]
You are not using the image you think you are. I showed you that the `rabbitmq:latest` image doesn't contain this plugin at all. I'm pretty sure it never has.

**Reply by michaelklishin** [reply]
What leads you to believe that? It's a community plugin that the image does not include, and never has IIRC.

**Reply by KamranShahid** [reply]
> What leads you to believe that? It's a community plugin that the image does not include, and never has IIRC.

Because it is the only image i have pull. no other container running. so what ever come i thought is within the image

**Reply by michaelklishin** [reply]
That's our point: the plugin you attempt to use is not included into the RabbitMQ distribution and is not included by the image. You have to find or build a new image that provisions and [pre-enables](https://rabbitmq.com/plugins.html#enabled-plugins-file) all plugins that do not ship with RabbitMQ.

**Comment by KamranShahid** [comment]
what i see is i have following line in volume
./rabbitmq/plugins/rabbitmq_delayed_message_exchange-3.8.9-0199d11c.ez:/opt/rabbitmq/plugins/rabbitmq_delayed_message_exchange-3.8.9-0199d11c.ez:rw

**Reply by mkuratczyk** [reply]
We've said it a few times already: https://hub.docker.com/_/rabbitmq doesn't contain this plugin at all. If you have issues related to this plugin, you are using a different image and it's this image's maintainer job to update the plugin.

Either way, when reporting an issue, please provide the steps to reproduce. Containers were invented for repeatability, so let's take advantage of that and provide `docker run ...` commands that lead to the issue (with SHA tags)
